core/src/main/kotlin/io/specmatic/core/SpecmaticConfig.kt

```diff
@@ -278,26 +299,46 @@ data class SpecmaticConfig(
 }
 
 data class TestConfiguration(
-    val resiliencyTests: ResiliencyTestsConfig? = ResiliencyTestsConfig(
-        isResiliencyTestFlagEnabled = getBooleanValue(SPECMATIC_GENERATIVE_TESTS),
-        isOnlyPositiveFlagEnabled = getBooleanValue(ONLY_POSITIVE)
-    ),
-    val validateResponseValues: Boolean? = null,
-    val allowExtensibleSchema: Boolean? = null,
-    val timeoutInMilliseconds: Long? = getLongValue(SPECMATIC_TEST_TIMEOUT)
-)
+    private val resiliencyTests: ResiliencyTestsConfig? = null,
+    private val validateResponseValues: Boolean? = null,
+    private val allowExtensibleSchema: Boolean? = null,
+    private val timeoutInMilliseconds: Long? = null
+) {
+    fun getResiliencyTests(): ResiliencyTestsConfig {
```

Comment:
In most cases, these getter methods on inner classes aren't necessary, as the objective of hiding the internal config structure from the rest of the application is achieved by the top-level method in SpecmaticConfig which returns a scalar.

However there are some cases in which the top-level method returns an object. In such cases, we would need these getter methods on the object being returned.